### PR TITLE
🍕Force search cursor to end.

### LINKED ISSUE #147: Bug while typing in search bar
When I type a letter in the search bar at the main page, it instantly shows the result. The new search bar in the navigation already has the focus, but the cursor is at the first and not at the last position.

When I search for example for "paper", I type in the "p", the cursor is at the first position, I type in the "aper" and my final search query is "aperp".

That's really annoying. Is it possible to set the cursor position with JS? I used the latest Firefox version ...

### PR DESCRIPTION
**Pull request by samccone**
Fixes #147

:bug: Bug

##### How to reproduce 

* visit http://elements.polymer-project.org in firefox :fire: :wolf: 
* type in top search bar "goat" :goat: 
* you should see `oatg` now in the search box on the left :crying_cat_face: 

##### What this fixes

* when you now type goat :goat: 
* the left input now says :goat: yay :hand: 

:sparkles:

### CODE REVIEW

File: app/elements/pages/page-browse.html [hunk 1]
```diff
@@ -238,7 +238,15 @@ <h2 class="flex"><span>[[pageTitle]]</span> <span class="count">[[_elementCount]
       updateSearch: function() {
         var params = this._parseQueryString();
         if (params && params.q) {
-          this.async(function() { this.$.query.focus(); });
+          this.async(function() {
+            this.$.query.focus();
+
+            // make sure to force the cursor to the end of the
+            // input. Only an issue in Firefox.
+            if (this.$.query.setSelectionRange) {
```
Comment (samccone) [inline app/elements/pages/page-browse.html]: not sure what kinda support range we want to support here... was just safe but we can :skull: this if need be

Comment (frankiefu) [inline app/elements/pages/page-browse.html]: Yeah I think `setSelectionRange` (basic) is well supported in all the browsers that Polymer supports.  We can probably remove the check here.

Comment (samccone) [inline app/elements/pages/page-browse.html]: Ok 🐸

**Comment by samccone** [comment]
all updated :cat:

**Comment by frankiefu** [comment]
LGTM :+1:

**Comment by addyosmani** [comment]
⚡⚡⚡

**Pull request merged**